### PR TITLE
doc: config_and_build: regroup HW and board pages

### PR DESCRIPTION
**Pull request by greg-fer**
Created "Configuring application" section with "Configuring devicetree". Moved devicetree related information under the devicetree section. The landing page reuses contents of Changing hardware config section.

Created "Board support" section.
Split information on former board support page into several pages.

NCSDK-22324.

### CODE REVIEW

**Comment by greg-fer** [comment]
@richabp , can you have a look at the shield table? I've added mentions of Wi-Fi boards there.

File: doc/nrf/device_guides/working_with_nrf/nrf70/gs.rst [hunk 1]
```diff
@@ -263,6 +263,8 @@ Shields and expansion boards
 
 Shields and expansion boards are add-on hardware that can be attached to a development kit or prototyping platform to extend their features and functionalities.
 
+.. _nrf70_gs_shields_expansion_boards_nrf7002ek:
```
Comment (richabp) [inline doc/nrf/device_guides/working_with_nrf/nrf70/gs.rst:266]:
```suggestion
.. _nrf70_gs_shields_evaluation_kit_nrf7002ek:
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Review by DeidreCasey — CHANGES_REQUESTED** [review]

File: doc/tags.yml [hunk 2]
```diff
@@ -36,6 +36,16 @@ mapping_topics:
                                   "thingy53", "nrf52840", "nrf52833", "nrf52832", "nrf52820",
                                   "nrf52811", "nrf52810", "nrf52805", "nrf21540", "npm1100",
                                   "npm1300", "npm6001"]
+  - nrf/config_and_build/configuring_app/*.html: ["applications", "samples", "development-kits",
+                                                  "evaluation-kits", "prototyping-platforms"]
+  - nrf/config_and_build/board_support/*.html: ["nrf91-series", "nrf70-series", "nrf53-series",
+                                                "nrf52-series", "nrf7002", "nrf7001", "nrf7000",
+                                                "nrf9160", "thingy91", "nrf5340", "thingy53",
```
[the remark this answers is not in the document]
Comment (richabp) [reply doc/tags.yml:43]: FYI, Uma has raised a PR for adding nrf9161: https://github.com/nrfconnect/sdk-nrf/pull/13591.

**Review by DeidreCasey — APPROVED** [review]

**Comment by richabp** [comment]
> @richabp , can you have a look at the shield table? I've added mentions of Wi-Fi boards there.

Confirmed with Bansi, the entries are correct.

**Review by richabp — APPROVED** [review]

**Pull request merged**